### PR TITLE
Feature/39517

### PR DESCRIPTION
**Pull request by abhipunj0812**
validation.js to validate Email users.
permissions.js to validate permission for different roles.

### CODE REVIEW

File: extra/utils/permissions.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            }
+            else if(permissionType=="Delete"){
+                f=Delete.includes(role)
+                return f;
+            }
+
+        }
+
+    }
+    d = hasPermission(getUsers,"trainer","Delete");
```
Comment (rahulsainidev) [inline extra/utils/permissions.js]: Here first parameter (module name) need to be passed as string you have passed as object. Fix this.

Comment (abhipunj0812) [reply extra/utils/permissions.js]: ok sir, I'll fix this and update you once done

**Review by rahulsainidev — COMMENTED** [review]
In validation.js the valid pair is that who has both the emails valid (reviewerEmail and traineeEmail). 
For example:
**Valid:** 
   ```
{
    traineeEmail: 'trainee1@successive.tech',
    reviewerEmail: 'reviewer1@successive.tech',
    },
```

**Invalid**
```
{
    traineeEmail: 'trainee2@successive.tech',
    reviewerEmail: 'reviewer2@succesive1.tech',
    }
```

**Review by rahulsainidev — COMMENTED** [review]
Code for checking permission in permissions.js can be done without loop by accessing moduleName and its permissions using ( [] ) bracket notation. Try to implement that.

Note: Use proper indentation so code will be easier to read.

**Review by rahulsainidev — APPROVED** [review]

**Pull request merged**